waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -67,16 +61,46 @@ genJobs spec = return $ genJob <$> getJobs spec
                   [ "jobName" .= jobName,
                     "jobPerformFnName" .= jobPerformFnName,
                     "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
-                    "jobFactoryName" .= show (jobFactoryForJob job)
+                    "jobFactoryName" .= show (jobFactoryForJob $ executor job),
+                    "jobPerformOptions" .= jobPerformOptionsJsonString job
                   ]
             )
 
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+jobFactoryForJob :: JobExecutor -> JobFactory
+jobFactoryForJob Passthrough = PassthroughJobFactory
+jobFactoryForJob PgBoss = PgBossJobFactory
+
 genJobFactories :: Generator [FileDraft]
 genJobFactories = return $ genJobFactory <$> jobFactories
   where
     genJobFactory :: JobFactory -> FileDraft
     genJobFactory jobFactory =
       let jobFactoryFp = jobFactoryFilePath jobFactory
           sourceTemplateFp = C.asTmplFile jobFactoryFp
-          destinationServerFp = C.asServerFile jobFactoryFp
-       in C.mkTmplFdWithDstAndData sourceTemplateFp destinationServerFp Nothing
+       in C.mkTmplFd sourceTemplateFp
+
+jobFactories :: [JobFactory]
+jobFactories = enumFrom minBound :: [JobFactory]
+
+jobFactoryFilePath :: JobFactory -> Path' (Rel d) File'
+jobFactoryFilePath PassthroughJobFactory = [relfile|src/jobs/PassthroughJobFactory.js|]
+jobFactoryFilePath PgBossJobFactory = [relfile|src/jobs/PgBossJobFactory.js|]
+
+isPgBossUsed :: AppSpec -> Bool
```

Comment:
Yep, so we can certainly move this one to AppSpec! And rename it to `isPgBossJobExecutorUsed`.